### PR TITLE
INTL JDI | Mobile | IN & UK | Template Carousel - Updates

### PR DESCRIPTION
**Pull request by jsandland**
Fixes following issues|Adds following features:
- swipe on mobile maintains the same behavior seen when using the left and right chevron navigation
- the number of cards scrolled through is relevant to number of displayed cards
- scroll from left to right snaps
- resizing browser maintains mobile / desktop behavior without refresh
- CSS loads for every carousel that gets loaded on the page
- every instance of a carousel on the page maintains it's own currentIndex
- every instance of a carousel on mobile starts at the second card 
- top and bottom of template-x template expands on hover
Resolves: [MWPW-155657](https://jira.corp.adobe.com/browse/MWPW-155657)

**Pages to check for regression and performance:**
- https://basic-carousel-updates--express--adobecom.hlx.page/drafts/jsandlan/mothers-day

### CODE REVIEW

**Review by JingleH — APPROVED** [review]

**Review by echen-adobe — APPROVED** [review]

**Review by JingleH — COMMENTED** [review]
the mobile interactions seem to be a little off

File: express/blocks/shared/basic-carousel.css [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  flex-wrap: nowrap;
+  overflow-y: hidden;
+  overflow-x: scroll;
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+  scroll-snap-type: x mandatory;
+  scroll-behavior: smooth;
+  max-height: 500px;
+  padding-top: 4px;
+  padding-bottom: 6px;
```
Comment (fullcolorcoder) [inline express/blocks/shared/basic-carousel.css:26]: This is difficult to read and maintain. Please group these together by something like:
- Display / flex / grid 
- Box model properties 
- Position/layout
- Typography
- Other styles like opacity and what not 

I don't have a major nit on what order things are in, as long as they're together. In other sections I'll mention to organize, meaning this.

File: express/blocks/shared/basic-carousel.css [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  display: flex;
+  position: absolute;
+  width: 53px;
+  align-items: center;
+  bottom: -40px;
+  right: 45px;
+  transition: opacity 0.2s, display 0.2s;
+  cursor: pointer;
+  -webkit-tap-highlight-color: rgba(0,0,0,0);
+  z-index: 1;
```
Comment (fullcolorcoder) [inline express/blocks/shared/basic-carousel.css:52]: Organize

File: express/blocks/shared/basic-carousel.css [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  position: absolute;
+  width: 53px;
+  align-items: center;
+  bottom: -40px;
+  right: 0;
+  flex-direction: row-reverse;
+  transition: opacity 0.2s;
+  cursor: pointer;
+  -webkit-tap-highlight-color: rgba(0,0,0,0);
+  z-index: 1;
```
Comment (fullcolorcoder) [inline express/blocks/shared/basic-carousel.css:67]: Organize

**Review by fullcolorcoder — CHANGES_REQUESTED** [review]
Looking pretty good. I have a few CSS asks and a few interaction/accessibility asks as well.

On mobile:
- Clicking on the next greyed out box doesn’t advance, which is inconsistent with other carousel implementations  

Accessibility: 
- Using the right arrow advances the carousel, not the selected card
- Lacking aria labels for nav elements

**Review by fullcolorcoder — APPROVED** [review]

**Pull request merged**